### DISCUSSION TITLE
Error while starting meershaum

### DISCUSSION THREAD

**Discussion by rohitsaikiran91** [bmeares/Meerschaum#65] [Q&A] [answered]
Hi, 
I am following the video tutorials. In Episode I the Basics, when I type in `show pipes` after `mrsm`, I get the following error.

ERROR (psycopg2.OperationalError) connection to server at "localhost" (::1), port 5432 failed: FATAL:  password
authentication failed for user "mrsm"

(Background on this error at: https://sqlalche.me/e/14/e3q8)

I know it's related to authentication, but I am not sure how to resolve.
Please let me know if there is any work around/ corrective action

**Comment by bmeares** [comment, marked as answer]
Hi @rohitsaikiran91, thanks for reaching out! Are you connecting to an existing database or trying to spin up the default database? Here are a couple options from the [Getting Started guide](https://meerschaum.io/get-started/):
- **Built-in SQLite database** (`sql:local`)  
  If you specify `sql:local` as your instance (e.g. `mrsm show pipes -i sql:local`), you can start using Meerschaum with a pre-configured SQLite database.
- **Default TimescaleDB database** (`sql:main`)  
  When you first run `mrsm stack up -d db`, Meerschaum starts a TimescaleDB database with the credentials specified under `sql:main`:

  ```yaml
  connectors:
    sql:
      main:
        username: mrsm
        password: mrsm
        flavor: timescaledb
        host: localhost
        database: meerschaum
        port: 5432
  ```

  If you change the password later on with [`ALTER ROLE`](https://www.postgresqltutorial.com/postgresql-administration/postgresql-change-password/), you can edit the credentials for `sql:main` with `mrsm edit config`.

  Omitting `--instance` or `-i` assumes your configured default instance (`sql:main`).

  To completely reset the Meerschaum stack, run `mrsm stack down -v` to stop the containers and remove the volumes.
- **Existing database**  
  You can add a new connector to an existing database with `mrsm bootstrap connector` (e.g. `sql:mydb`). You can set this new connector as your default instance by changing the value of `instance` to your new connector with `mrsm edit config`:

  ```yaml
  instance: sql:mydb
  ```

I hope this helps! If you are looking to use the default stack, you can reset everything with `mrsm stack down -v` and `mrsm stack up -d db`. Otherwise you will need to specify a different instance to your chosen database (in the interactive shell, the `instance` command will append `--instance` to every command, e,g. `instance sql:local`). Let me know if you are still running into issues!

**Reply by rohitsaikiran91** [reply]
I am trying to use the default timescaledb. I haven't made any changes to `config` using `mrsm edit config`. 

I ran the command `mrsm stack up -d db grafana` followed by `mrsm` which opened the meershaum CLI. I then ran the command `show pipes` . I get this error:


I am running on Win 10 using a python venv
I have PGAdmin and postgress 14 installed but I am not running it. My assumption is that `db` is a docker container that runs postgress server on `PORT:5432` of host to `PORT:5432` of the container.

```u:\python_practice\meershaum\venv\lib\site-packages\meerschaum\connectors\sql\tables\__init__.py:171: UserWarning:  WARNING (psycopg2.OperationalError) connection to server at "localhost" (::1), port 5432 failed: FATAL:  password
authentication failed for user "mrsm"~~\ \__/  |  |
(venv) U:\python_practice\meershaum\venv\Scripts>mrsm
(Background on this error at: https://sqlalche.me/e/14/e3q8)
  warn(str(e))
 [ mrsm@sql:main ] > show pipes
 -
 ERROR (psycopg2.OperationalError) connection to server at "localhost" (::1), port 5432 failed: FATAL:  password
authentication failed for user "mrsm"

(Background on this error at: https://sqlalche.me/e/14/e3q8)
```

I tried `stack down -v ` and restarted using `stack up -d db grafana`. I still get the same error. 

I tried going to the bash terminal of `db` docker container to change the authentication using `ALTER ROLE` but I was not able to make any changes as it was showing the following error:

```psql: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "root" does not exist```

I tried `psql -U mrsm` but the same error.

Not sure where's the exact issue

**Reply by bmeares** [reply]
Hm, I can't reproduce the issue on my Windows machine, so I think it might be your environment. Are you running your containers through Docker Desktop? E.g. after running `mrsm stack up -d db grafana`, does Docker Desktop look like this?

![Docker Desktop](https://i.imgur.com/KQR4EZW.png)

Perhaps your other PostgreSQL container might still be running on port 5432 and intercepting the requests. Try verifying everything is down (I had to do a reboot after installing Docker Desktop because I had containers running via `dockerd` on WSL) before bringing back up the stack. If you have WSL set up, you could also try Meerschaum from the Linux environment.

The default configuration directory on Windows is `%APPDATA%\Meerschaum` (Linux / MacOS: `~/.config/meerschaum/`). Remove this directory if you're trying to completely reset your environment.

**Reply by bmeares** [reply]
Hi @rohitsaikiran91, were you able to get your stack running?

**Reply by rohitsaikiran91** [reply]
Hi Sorry for the late reply. So I have uninstalled PG-admin and postgress (Although nothing was running on `PORT: 5432`) just to make sure there is no interference. I deleted the meerschaum folder in `%APPDATA%\Roaming\Meerschaum` And then tried, its' working now. Not sure what the issue was earlier. I am gonna install pg-admin and postgress to check if this is what's causing the issue.
